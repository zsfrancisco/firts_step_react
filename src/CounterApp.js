import React, { useState } from 'react';
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);


    const handleAdd = (e) => {
        setCounter(counter + 1);
        // Tabmbién se puede de la siguiente forma:
        // setCounter( (c) => c + 1 );
    }
    const handleSub = (e) => {
        setCounter(counter - 1);
        // Tabmbién se puede de la siguiente forma:
        // setCounter( (c) => c + 1 );
    }
    const handleReset = (e) => {
        setCounter(value);
        // Tabmbién se puede de la siguiente forma:
        // setCounter( (c) => c + 1 );
    }

    return (
        <>
            <h1>CounterApp</h1>
            <p>{ counter}</p>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleReset}>Reset</button>
            <button onClick={handleSub}>-1</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}

export default CounterApp;