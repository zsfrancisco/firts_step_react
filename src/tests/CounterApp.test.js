import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import React from 'react';
import CounterApp from '../CounterApp';

describe('Pruebas sobre el counter app', () => {
    const value = 10;
    let wrapper = shallow( <CounterApp value={ value } /> );;

    // Función que se ejecuta antes de cada prueba, en este caso, se la usa para
    // reiciar el contador.
    beforeEach(() => {
        wrapper = shallow( <CounterApp value={ value } /> );
    });

    test('Debe mostrar <CounterApp /> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar el valor por defecto de 100', () => {
        const defaultValue = wrapper.find('p').text();
        // console.log('defaultValue: ', defaultValue);
        expect(defaultValue).toBe(value.toString());
    });
    
    test('Debe aumentar el contador en +1', () => {
        // Primer botón
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('p').text();
        expect(counterText).toBe('11');
    });
    
    test('Debe resetear el contador en reset', () => {
        // Primer botón
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('p').text();
        expect(counterText).toBe('10');
    });
    
    test('Debe decrementar el contador en -1', () => {
        // Primer botón
        const buton = wrapper.find('button').at(2).simulate('click');
        // console.log('botón: ', buton.html());
        const counterText = wrapper.find('p').text();
        expect(counterText).toBe('9');
    });
    

})
