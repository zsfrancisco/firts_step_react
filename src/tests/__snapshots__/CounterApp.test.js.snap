// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el counter app Debe mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <p>
    10
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
